manager/integration/tests/test_recurring_job.py

```diff
@@ -724,25 +716,23 @@ def test_recurring_jobs_when_volume_detached_unexpectedly(settings_reset, set_ra
         }
     ]
     vol.recurringUpdate(jobs=jobs)
-
-    wait_for_recurring_backup_to_start(client,
-                                       core_api,
-                                       vol_name,
-                                       expected_snapshot_count=1)
+    time.sleep(60)
+    wait_for_recurring_backup_to_start(client, core_api, vol_name,
+                                       expected_snapshot_count=1,
+                                       minimum_progress=50)
 
     crash_engine_process_with_sigkill(client, core_api, vol_name)
-    # waiting 30sec for volume detach and attach operation
-    # after recurring backup is interrupted
-    time.sleep(30)
+    # Check if the volume is reattached after recurring backup is interrupted
+    time.sleep(10)
     wait_for_volume_healthy_no_frontend(client, vol_name)
 
-    snapshot_name = \
-        wait_for_recurring_backup_to_start(client,
-                                           core_api,
-                                           vol_name,
-                                           expected_snapshot_count=1)
+    # Since the backup state is removed after the backup complete and it
+    # could happen quickly. Checking for the both in-progress and complete
+    # state could be hard to catch, thus we only check the complete state
+    def backup_failure_predicate(b):
```

Comment:
I guess this should be called backup_success_predicate?

Reply:
Fixed.